apk-generator/v2/app/generator/generator.py

```diff
@@ -155,6 +156,12 @@ def generate(self, should_notify=True):
         replace(self.get_path("app/src/main/res/layout/nav_header.xml"), 'twitter', 'background')
         replace(self.get_path("app/build.gradle"), '"org.fossasia.openevent"', '"%s"' % self.app_package_name)
 
+        if self.theme_colors:
+            default_colors = colors()
+            self.update_status('Setting theme colors')
+            for key in default_colors.keys():
+                replace(self.get_path("app/src/main/res/values/color.xml"), default_colors[key], self.theme_colors[key])
```

Comment:
Instead of searching for color and replacing, wouldn't it be better to parse the xml and change colors by key